Fixes #2911 - Update modals

munteanu210 opened the pull request:
<!--
IMPORTANT: Please do not create a Pull Request without creating an issue first. 

Read the Guidelines, If you haven't done it yet.
https://github.com/webcompat/webcompat.com/blob/master/docs/pr-coding-guidelines.md
-->

This PR fixes issue #2911

## Proposed PR background


This pull request fixes issue #2911 Update modals. The pull request has the front end development for the following two pop-ups:

* [x]  What is web compatibility (it also includes a slider). See [screenshot](https://www.screencast.com/t/HEbLgpMD7eyl). 
* [x]  Privacy Policy. See [screenshot](https://www.screencast.com/t/Jvh7xmZ3Sr06).

Here is the URL to miro design for these two pop-ups: https://miro.com/app/board/o9J_kxWDuVE=/

johngian commented on webcompat/templates/layout.html:
```diff
@@ -36,7 +36,14 @@
 <script src="{{ url_for('static', filename='js/lib/bugform.js') }}"></script>
+
+{%- if ab_active('exp') == 'form-v2' %}
+<script src="{{ url_for('static', filename='js/lib/popup.js') }}"></script>
```
Please use the same name convention we decided to keep with the rest of the filenames eg:
`issue-wizard-popup.js`
`issue-wizard-slider.js`

Also is there any reason not to combine them to one file just to avoid making things even more complicated with the 2 different codebases?

johngian commented:
Is there any reason `sprite.svg` is changing in this PR?

johngian commented:
Also where are you importing the `popup.css` and `slider.css`  assets?

johngian commented on webcompat/templates/home-page/issue-wizard-form.html:
```diff
@@ -21,14 +21,14 @@ <h3 class="page-heading">Filing a web compatibility issue</h3>
           <div class="form-text form-element js-Form-group form-label-message with-validation-icons">
             {{ form.url.label(class_='form-label is-hidden') }}
             {% if form.url.errors %}
-              {% for error in form.url.errors %}
```
Plase make sure you follow the same indentation paradigm across the file. This PR introduces a lot of indentation changes that are not really required nor correct.

johngian replied:
This also has other occurences

johngian commented on webcompat/templates/home-page/issue-wizard-form.html:
```diff
@@ -362,5 +364,74 @@ <h3 class="page-heading">Filing a web compatibility issue</h3>
+<div
+  class="popup-modal shadow"
+  data-popup-modal="what-is-webcompat">
+  <img class="close popup-modal__close" src="../../img/svg/icons/svg-cross-black.svg" title="Close popup" alt="Close popup" />
```
This `src` definition will be broken in the actual deployment. Please use the same static imports like we do in the rest of the template eg:
```
templates/home-page/issue-wizard-form.html:    <img class="warning-icon" src="{{ url_for('static', filename='img/svg/icons/svg-info-icon.svg') }}" title="Warning" alt="Warning" />

```

johngian replied:
This also has other occurences

johngian commented on webcompat/templates/home-page/issue-wizard-form.html:
```diff
@@ -362,5 +364,74 @@ <h3 class="page-heading">Filing a web compatibility issue</h3>
+            <div class="text">A web compatibility issue occurs <strong>when a website behaves differently across browsers.</strong> Sometimes a website will load on one browser and crash or render in a strange way on another. </div>
+          </div>
+          <div class="image-half">
+              <img class="" src="../../img/svg/help-step-1.svg" title="What is a web compatibility issue" alt="What is a web compatibility issue" />
```
Same here

johngian commented on webcompat/templates/home-page/issue-wizard-form.html:
```diff
@@ -362,5 +364,74 @@ <h3 class="page-heading">Filing a web compatibility issue</h3>
+            <div class="text">Sometimes websites have bugs that prevent them from working in all browsers. Browsers also have different ways of implementing features that may affect a site’s usability. </div>
+          </div>
+          <div class="image-half">
+              <img class="" src="../../img/svg/help-step-2.svg" title="What is a web compatibility issue" alt="What is a web compatibility issue" />
```
Same here.

johngian commented on webcompat/templates/home-page/issue-wizard-form.html:
```diff
@@ -362,5 +364,74 @@ <h3 class="page-heading">Filing a web compatibility issue</h3>
+            <div class="text"><strong>We want to make sure the Internet works for everyone.</strong> Together with web developers and site owners, we help identify and fix web compatibility issues. Let us know if you find a website behaves differently across browsers. </div>
+          </div>
+          <div class="image-half">
+              <img class="" src="../../img/svg/help-step-3.svg" title="What is a web compatibility issue" alt="What is a web compatibility issue" />
```
Same here.

johngian commented on webcompat/templates/home-page/issue-wizard-form.html:
```diff
@@ -362,5 +364,74 @@ <h3 class="page-heading">Filing a web compatibility issue</h3>
+<div
+  class="popup-modal shadow text"
+  data-popup-modal="privacy-modal">
+  <img class="close popup-modal__close" src="../../img/svg/icons/svg-cross-black.svg" title="Close popup" alt="Close popup" />
```
Same here

ksy36 commented on webcompat/static/js/lib/issue-wizard-bugform.js:
```diff
@@ -798,6 +803,10 @@ BugForm.prototype.nextStep = function(e) {
 
+    if (trigger.val() === this.otherProblemId) {
+      hideStep = 4;
+    }
```
This condition is part of `BugForm.prototype.nextStep` method that is quite big and contains lots of logic in it. Could you please split the method into smaller functions ?

ksy36 commented on webcompat/static/js/lib/issue-wizard-popup.js:
```diff
@@ -0,0 +1,119 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+function Popup() {
+  this.init = function() {
```
This class seems to be responsible for both popup and slider. Is there a way to separate them? Perhaps could create a separate Slider class?

munteanu210 replied:
@ksy36 Ksenia, the popup and slider were separate in my initial commit. @johngian asked me to combine them, which I did. See [screenshot](https://www.screencast.com/t/T34o99zOrCAF). Can you and @johngian discuss, and let me know what your common decision is?

karlcow replied:
@munteanu210 please use links to previous comment instead of screenshots.  The screenshot might go away and we will loose context. Links are first class citizen for discussions. 😉 

Here's the link. https://github.com/webcompat/webcompat.com/pull/2937#discussion_r327076243

for this text.
> Also is there any reason not to combine them to one file just to avoid making things even more complicated with the 2 different codebases?

PS: to get the exact link, you just need to click on the time information located after the name of the commenter. Hope it helps.

karlcow replied:
My impression for [@johngian comment](https://github.com/webcompat/webcompat.com/pull/2937#discussion_r327076243) is that it is about files name conventions and organizations, while @ksy36 is about logical code.

ksy36 replied:
Ah thanks for pointing this out, I have missed that comment @munteanu210.

So yeah I agree with your initial approach, when a single class has single responsibility it's easier to maintain later on. 

@johngian we could perhaps leave them in separate files? If you think it will complicate things, we could move all js files for the new form to a designated folder. Let me know what you thoughts are :)

johngian replied:
I don't have a strong opinion between single/multiple files. I just want to make sure that the naming is consistent enough so we can isolate the 2 codebases.

miketaylr commented:
(and please use the commit message convention going forward -- it's how we tie the history of the project to our github issues easily, and our CHANGELOG tooling relies on it)

munteanu210 commented:
@johngian Thank you for reviewing this PR and sending feedback. You can find my answers below:

> Is there any reason `sprite.svg` is changing in this PR?
The sprite.svg file was updated by the build:svg script. I can revert it if it doesn't belong here.

>Also where are you importing the popup.css and slider.css assets?
You are right. This is my omission. Accidentally, the import went to the next PR that I was about to do in the next days. I included the import to the commit that I did 30 mins ago in this PR.
 
>Also is there any reason not to combine them to one file just to avoid making things even more complicated with the 2 different codebases?
I went with your advice and merged these two components. popup.js and slider.js became popup.js.

>Please make sure you follow the same indentation paradigm across the file. This PR introduces a lot of indentation changes that are not really required nor correct.
I am a maniac of correct indentation, so we are on the same page on this. I don't know why the code got intended in this way. I have not touched this code in the last N commits. I went through the entire document and put the correct indentation.

>This src definition will be broken in the actual deployment. Please use the same static imports like we do in the rest of the template. This also has other occurrences. 
I changed the URLs for the images that you mentioned about. I found several other images with this same problem, and fixed them all.

I marked [all these items](https://www.screencast.com/t/Xh3iBWY8dqoF) as resolved.

miketaylr commented:
I'm gonna assume @johngian is OK with me merging this without his explicit signoff. If there are follow up issues, please file a bug!
